src/predicates/propPathSatisfies.js

```diff
@@ -0,0 +1,57 @@
+/** @license ISC License (c) copyright 2018 original and current authors */
+/** @author Ian Hofmann-Hicks (evilsoft) */
+
+const curry = require('../core/curry')
+const isArray = require('../core/isArray')
+const isDefined = require('../core/isDefined')
+const isEmpty  = require('../core/isEmpty')
+const isInteger = require('../core/isInteger')
+const isNil = require('../core/isNil')
+const isPredOrFunc = require('../core/isPredOrFunc')
+const isString = require('../core/isString')
+const predOrFunc = require('../core/predOrFunc')
+
+// propPathSatisfies: [ (String | Integer) ] -> (a -> Boolean) -> b -> Boolean
+// propPathSatisfies: [ (String | Integer) ] -> Pred a -> b -> Boolean
+function propPathSatisfies(keys, pred, x) {
+  if(!isArray(keys)) {
+    throw new TypeError(
+      'propPathSatisfies: Array of Non-empty Strings or Integers required for first argument'
+    )
+  }
+
+  if(!isPredOrFunc(pred)) {
+    throw new TypeError(
+      'propPathSatisfies: Pred or predicate function required for second argument'
+    )
+  }
+
+  if(isNil(x)) {
```

Comment:
Should we be throwing in this case? `if (!isObject(x)) { throw new TypeError( 'propPathSatisfies: Object required for third argument') } `

Reply:
No. because the path does not exist.
AND Predicates do not throw in crocks (or shouldn't) they either are or are not.